components/scream/scripts/nc_file_init.py

```diff
@@ -0,0 +1,359 @@
+        expect (close!=-1,"Error! Var declaration should be 'name(dim1,...,dimN)'.")
+        expect (close>open,"Error! Var declaration should be 'name(dim1,...,dimN)'.")
+
+        name = name_dims[0:open]
+        return name
+
+    ###########################################################################
+    def get_dims(self,name_dims):
+    ###########################################################################
+        open = name_dims.find('(')
+        close = name_dims.find(')')
+
+        # Check format
+        expect (open!=-1,"Error! Var declaration should be 'name(dim1,...,dimN)'.")
+        expect (close!=-1,"Error! Var declaration should be 'name(dim1,...,dimN)'.")
+        expect (close>open,"Error! Var declaration should be 'name(dim1,...,dimN)'.")
+        expect (close==len(name_dims)-1,"Error! Var declaration should be 'name(dim1,...,dimN)'.")
+
+        dims = name_dims[open+1:close].split(',')
+        expect (len(dims)>0,"Error! Var declaration should be 'name(dim1,...,dimN)'.")
+
+        return dims
+
+    ###########################################################################
+    def is_vector_layout(self,dims):
+    ###########################################################################
+        valid = ["COL", "LEV", "ILEV"]
+        for dim in dims:
+            if dim not in valid:
+                expect (dim.isdigit(), "Error! Unexpected dimension '{}'".format(dim))
+                return True
+        return False
+
+    ###########################################################################
+    def get_scalar_dims(self,dims):
+    ###########################################################################
+        valid = ["COL", "LEV", "ILEV"]
+        s_dims = []
+        vec_dim_id = -1 
+        for i in range(0,len(dims)):
+            if dims[i] in valid:
+                s_dims.append(dims[i])
+            else:
+                expect (vec_dim_id==-1,
+                        "Error! Multiple integer extents found in dims specification '{}'.\n"
+                        "       Only vectors are supported, for non-scalar layouts.".format(dims))
+                vec_dim_id = i
+
+        expect(vec_dim_id>0, "Error! Something went wrong while detecting vector dim id from '{}'.".format(dims))
+
+        return vec_dim_id, s_dims
+
+    ###########################################################################
+    def get_values(self,vals_str,vec_dim):
+    ###########################################################################
+        if vals_str=="":
+            # User did not specify values. Use all zeros.
+            return [0]*vec_dim
+        else:
+            try:
+                # Try to convert vals_str to a single float. If successful,
+                # the user passed a single value for all vector components
+                value = float(vals_str)
+                return [value]*vec_dim
+            except:
```

Comment:
Minor point, python recommends against bare except statements. You should catch the exception type that can arise through semi-normal execution (in this case, it's the failure to convert str->float which is a ValueError) while leaving more surprising exceptions uncaught.

Reply:
Good point.